addons/pos_restaurant/static/src/app/services/pos_store.js

```diff
@@ -257,8 +276,62 @@ patch(PosStore.prototype, {
             this.loadingOrderState = false;
         }
     },
+    editFloatingOrderName(order) {
+        this.dialog.add(EditOrderNamePopup, {
+            title: _t("Edit Order Name"),
+            placeholder: _t("18:45 John 4P"),
+            startingValue: order.floating_order_name || "",
+            getPayload: async (newName) => {
+                if (typeof order.id == "number") {
+                    this.data.write("pos.order", [order.id], {
+                        floating_order_name: newName,
+                    });
+                } else {
+                    order.floating_order_name = newName;
+                }
+            },
+        });
+    },
+    setFloatingOrder(floatingOrder) {
+        if (this.get_order()?.isDirectSale) {
+            this.transferOrder(this.get_order().uuid, null, floatingOrder);
+            return;
+        }
+        this.set_order(floatingOrder);
+
+        const props = {};
+        const screenName = floatingOrder.get_screen_data().name;
+        if (screenName === "PaymentScreen") {
+            props.orderUuid = floatingOrder.uuid;
+        }
+
+        this.showScreen(screenName || "ProductScreen", props);
+    },
+    searchOrder(buffer) {
+        const find_table = (t) => t.table_number === parseInt(buffer);
+        const table =
+            this.currentFloor?.table_ids.find(find_table) ||
+            this.models["restaurant.table"].find(find_table);
+        if (table) {
+            return this.setTableFromUi(table);
+        }
+        const floating_order = this.get_open_orders().find((o) => o.floatingOrderName === buffer);
+        if (floating_order) {
+            return this.setFloatingOrder(floating_order);
+        }
+        if (!table && !floating_order) {
+            const newOrder = this.add_new_order();
+            newOrder.floating_order_name = buffer;
+            newOrder.setBooked(true);
+            return this.setFloatingOrder(newOrder);
+        }
```

Comment:
I think when searching for an order by its table name should not create a floating order.